modules/nlu/src/backend/bootstrap.ts

```diff
@@ -10,22 +10,26 @@ import { BotFactory } from './application/bot-factory'
 import { BotService } from './application/bot-service'
 import { ScopedDefinitionsRepository } from './application/scoped/infrastructure/definitions-repository'
 import { ScopedModelRepository } from './application/scoped/infrastructure/model-repository'
-import { TrainingQueue } from './application/training-queue'
+import { TrainingQueue, TrainingQueueOptions } from './application/training-queue'
 import { TrainingRepository } from './application/training-repo'
 import { TrainingService } from './application/training-service'
 import { BotDefinition } from './application/typings'
 
 export async function bootStrap(bp: typeof sdk): Promise<NLUApplication> {
   const globalConfig: Config = await bp.config.getModuleConfig('nlu')
 
-  const { ducklingEnabled, ducklingURL, languageSources, modelCacheSize } = globalConfig
+  const { ducklingEnabled, ducklingURL, languageSources, modelCacheSize, maxTrainingPerInstance } = globalConfig
   const parsedConfig: sdk.NLU.Config = {
     languageSources,
     ducklingEnabled,
     ducklingURL,
     modelCacheSize: bytes(modelCacheSize)
   }
 
+  const trainingQueueConfig: Partial<TrainingQueueOptions> = {
+    ...(maxTrainingPerInstance && { maxTraining: maxTrainingPerInstance })
```

Comment:
why not just :

```ts
const trainingQueueConfig: Partial<TrainingQueueOptions> = {
    maxTraining: maxTrainingPerInstance
}
```
?

Reply:
can't maxTrainingPerInstance be undefined if older version configs are loaded? typescript partials only check for key existence not value types from what I know

Reply:
You're right! Had to double check just to make sure.

Well done to have thought about it

Reply:
Yo this is nice. I didn't even knew this syntax existed. There's [stack overflow post](https://stackoverflow.com/questions/11704267/in-javascript-how-to-conditionally-add-a-member-to-an-object/40560953#40560953) about it.